policybot/handlers/githubwebhook/filters/testresultfilter/testresultfilter.go

```diff
@@ -23,65 +23,56 @@ import (
 	"istio.io/bots/policybot/handlers/githubwebhook/filters"
 	"istio.io/bots/policybot/pkg/config"
 	gatherer "istio.io/bots/policybot/pkg/resultgatherer"
-	"istio.io/bots/policybot/pkg/storage"
 	"istio.io/bots/policybot/pkg/storage/cache"
 	"istio.io/pkg/log"
 )
 
 // Updates the DB based on incoming GitHub webhook events.
-type ResultGatherer struct {
-	store              storage.Store
-	repos              map[string]bool
-	cache              *cache.Cache
-	testResultGatherer *gatherer.TestResultGatherer
+type TestResultFilter struct {
+	repos map[string]gatherer.TestResultGatherer
+	cache *cache.Cache
+	// testResultGatherer *gatherer.TestResultGatherer
 }
 
 var scope = log.RegisterScope("ResultGatherer", "Result gatherer for each pr test run", 0)
 
-func NewResultGatherer(store storage.Store,
-	cache *cache.Cache, orgs []config.Org, bucketName string) filters.Filter {
+func NewTestResultFilter(cache *cache.Cache, orgs []config.Org, bucketName string) filters.Filter {
 	ctx := context.Background()
 
 	client, err := s.NewClient(ctx)
-	if err != nil {
-		return nil
-	}
-	testResultGatherer, err := gatherer.NewTestResultGatherer(client, bucketName)
 	if err != nil {
 		scope.Errorf(err.Error())
 		return nil
 	}
-	r := &ResultGatherer{
-		store:              store,
-		repos:              make(map[string]bool),
-		cache:              cache,
-		testResultGatherer: testResultGatherer,
+
+	r := &TestResultFilter{
+		repos: make(map[string]gatherer.TestResultGatherer),
+		cache: cache,
 	}
 
 	for _, org := range orgs {
 		for _, repo := range org.Repos {
-			r.repos[org.Name+"/"+repo.Name] = true
+			r.repos[org.Name+"/"+repo.Name] = gatherer.TestResultGatherer{client, bucketName, org.PreSubmitTestPath, org.PostSubmitTestPath}
```

Comment:
To be consistent, I think you should get the bucket name from the org struct, rather than having it come in explicitly.